vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -139,7 +140,8 @@ class OnboardingViewModel @AssistedInject constructor(
             is OnboardingAction.UpdateServerType           -> handleUpdateServerType(action)
             is OnboardingAction.UpdateSignMode             -> handleUpdateSignMode(action)
             is OnboardingAction.InitWith                   -> handleInitWith(action)
-            is OnboardingAction.UpdateHomeServer           -> handleUpdateHomeserver(action).also { lastAction = action }
+            is OnboardingAction.SelectHomeServer           -> run { lastAction = action }.also { handleHomeserverChange(action.homeServerUrl) }
```

Comment:
actions are set straight away as setting them afterwards implicitly relies on asynchronous coroutines (which isn't the case for tests!)